engineio/context.go

```diff
@@ -0,0 +1,33 @@
+package engineio
+
+import (
+	"context"
+	"io"
+	"net/http"
+
+	"github.com/googollee/go-socket.io/engineio/packet"
+)
+
+// Context provides info for a HTTP request.
+type Context struct {
+	// Standard context
+	context.Context
+
+	// The instance of current session.
+	Session Session
+
+	// The http.Request of current HTTP request.
+	// It may be different when using polling transport.
+	Request *http.Request
+
+	// The current packet infomation.
+	Packet Packet
```

Comment:
why are we need public access for context packet?

Reply:
`OnMessage()` needs to access `Context.Packet.Body` to get message body.

Reply:
i wrote some think at next comment